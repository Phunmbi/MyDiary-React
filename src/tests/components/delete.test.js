import React from 'react';
import { mount } from 'enzyme';
import { MemoryRouter } from 'react-router-dom';
import toJson from 'enzyme-to-json';
import DeleteComponent, { Delete } from '../../components/Delete';
import Root from '../../root';

let wrapper;
let wrapped;

beforeEach(() => {
  jest.useFakeTimers();
  wrapper = mount(
    <Root>
      <MemoryRouter initialEntries={[{ key: 'testkey' }]}>
        <DeleteComponent />
      </MemoryRouter>
    </Root>,
  );

  wrapped = shallow(
    <Delete />,
  );
});

afterEach(() => wrapper.unmount());

describe('Delete UI', () => {
  describe('render features', () => {
    test('component should render as expected', () => {
      const tree = toJson(wrapper);
      expect(tree).toMatchSnapshot();
    });
  });
});

describe('Delete Functionality', () => {
  it('should click yes', () => {
    const inst = wrapped.instance();

    jest.runAllTimers();

    const deleteEntry = jest.fn();
    inst.props = {
      match: {
        params: 1,
      },
      deleteEntry,
    };

    const yesButton = wrapped.find('#yes-button');
    yesButton.simulate('click');
  });

  it('should click No', () => {
    const inst = wrapped.instance();

    inst.setState({});

    const deleteEntry = jest.fn();
    inst.props = {
      history: {
        goBack: jest.fn(),
      },
      deleteEntry,
    };

    const yesButton = wrapped.find('#no-button');
    yesButton.simulate('click');
  });
});
