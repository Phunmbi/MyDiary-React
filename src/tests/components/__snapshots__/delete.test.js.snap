// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Delete UI render features component should render as expected 1`] = `
<_default>
  <Provider
    store={
      Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      }
    }
  >
    <MemoryRouter
      initialEntries={
        Array [
          Object {
            "key": "testkey",
          },
        ]
      }
    >
      <Router
        history={
          Object {
            "action": "POP",
            "block": [Function],
            "canGo": [Function],
            "createHref": [Function],
            "entries": Array [
              Object {
                "hash": "",
                "key": "testkey",
                "pathname": "/",
                "search": "",
              },
            ],
            "go": [Function],
            "goBack": [Function],
            "goForward": [Function],
            "index": 0,
            "length": 1,
            "listen": [Function],
            "location": Object {
              "hash": "",
              "key": "testkey",
              "pathname": "/",
              "search": "",
            },
            "push": [Function],
            "replace": [Function],
          }
        }
      >
        <Connect(Delete)>
          <Delete
            deleteEntry={[Function]}
            response={
              Object {
                "auth": Object {
                  "authenticated": null,
                  "firstName": null,
                  "lastName": null,
                },
                "entries": Object {
                  "authenticated": "",
                  "deletedEntry": Array [],
                  "deletedEntryError": Array [],
                  "deletedReminder": Array [],
                  "deletedReminderError": Array [],
                  "editedEntry": Array [],
                  "editedEntryError": Array [],
                  "entries": Array [],
                  "entriesError": Array [],
                  "error": Array [],
                  "existingReminder": Array [],
                  "existingReminderError": Array [],
                  "firstName": "",
                  "lastName": "",
                  "newEntry": Array [],
                  "newEntryError": Array [],
                  "newReminder": Array [],
                  "newReminderError": Array [],
                  "singleEntry": Array [],
                  "singleEntryError": Array [],
                  "status": "",
                },
                "profile": Object {
                  "authenticated": "",
                  "deletedEntry": Array [],
                  "deletedEntryError": Array [],
                  "deletedReminder": Array [],
                  "deletedReminderError": Array [],
                  "editedEntry": Array [],
                  "editedEntryError": Array [],
                  "entries": Array [],
                  "entriesError": Array [],
                  "error": Array [],
                  "existingReminder": Array [],
                  "existingReminderError": Array [],
                  "firstName": "",
                  "lastName": "",
                  "newEntry": Array [],
                  "newEntryError": Array [],
                  "newReminder": Array [],
                  "newReminderError": Array [],
                  "singleEntry": Array [],
                  "singleEntryError": Array [],
                  "status": "",
                },
              }
            }
          >
            <div
              className="dashboard"
            >
              <div
                id="deleteModal"
              >
                <div
                  className="deleteModal-main"
                >
                  <div
                    className="deleteModal-question"
                  >
                    <p
                      id="response"
                    />
                    <p>
                      Are you sure you want to delete this entry?
                    </p>
                  </div>
                  <div
                    className="deleteModal-button"
                  >
                    <button
                      className="button"
                      id="yes-button"
                      onClick={[Function]}
                      type="button"
                    >
                      Yes
                    </button>
                    <button
                      className="button"
                      id="no-button"
                      onClick={[Function]}
                      type="button"
                    >
                      No
                    </button>
                  </div>
                </div>
              </div>
            </div>
          </Delete>
        </Connect(Delete)>
      </Router>
    </MemoryRouter>
  </Provider>
</_default>
`;
